### DISCUSSION TITLE
HTTP error in runtime

### DISCUSSION THREAD

**Discussion by johnbuts** [mindee/doctr#1430] [Q&A]
Hello! Hope whoever finds this is doing well, I'm trying to use doctr on a container, and am able to get it downloaded no issues (via pip), but whenever I try to run it I am faced with the following error message. It's important to note that I am within a firewall and the http connections are supposed to be halted. I was wondering if there was a work around I could do since I have http/internet connection earlier in my dockerfile, could I curl the zip file (db_resnet50-adcafc63.zip)  its asking for and put it into the /home/appuser/.cache/doctr/models/ directory beforehand? Or is there something else I'm missing. Thanks in advanced! 

Error:

Calling doctr...
2024-01-16 16:09:52.621217: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-01-16 16:09:52.623674: I external/local_tsl/tsl/cuda/cudart_stub.cc:31] Could not find cuda drivers on your machine, GPU will not be used.
2024-01-16 16:09:52.652659: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-01-16 16:09:52.652693: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-01-16 16:09:52.653596: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-01-16 16:09:52.658634: I external/local_tsl/tsl/cuda/cudart_stub.cc:31] Could not find cuda drivers on your machine, GPU will not be used.
2024-01-16 16:09:52.659306: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-01-16 16:09:53.376743: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
Downloading https://doctr-static.mindee.com/models?id=v0.2.0/db_resnet50-adcafc63.zip&src=0 to /home/appuser/.cache/doctr/models/db_resnet50-adcafc63.zip
Failed download. Trying https -> http instead. Downloading http://doctr-static.mindee.com/models?id=v0.2.0/db_resnet50-adcafc63.zip&src=0 to /home/appuser/.cache/doctr/models/db_resnet50-adcafc63.zip
*************** DOCTR EXCEPTION ******************
HTTP Error 407: Proxy Authentication Required

**Comment by felixT2K** [comment]
Hi @johnbuts 👋🏼,

Sure you can download the model before the url is:
PyTorch: https://doctr-static.mindee.com/models?id=v0.3.1/db_resnet50-ac60cadc.pt&src=0
TF: https://doctr-static.mindee.com/models?id=v0.2.0/db_resnet50-adcafc63.zip&src=0

We provide also predefined docker images see: https://github.com/mindee/doctr/pkgs/container/doctr

If there is a problem with the default cache dir you could also modify it by: https://mindee.github.io/doctr/using_doctr/running_on_aws.html

**Reply by johnbuts** [reply]
Thanks for the swift response! So I downloaded the zip file beforehand, and changed the cache dir environmental variable to the appropriate location, and doctr is still making an https call to the url (https://doctr-static.mindee.com/models?id=v0.2.0/db_resnet50-adcafc63.zip&src=0) and causing an error. Is there a way to get around this by not changing the source code on my end?

**Reply by johnbuts** [reply]
Just spit balling here, I think the codebase is downloading the zip file every time it's ran, even if the zip exists in the target folder (which may lead to performance issues).

**Reply by felixdittrich92** [reply]
@johnbuts ah yeah sure :sweat_smile: 
you can pass:

```
pretrained=False,
pretrained_backbone=False,
```
to the model instance and to the `ocr_predictor` instance to avoid this

**Reply by johnbuts** [reply]
So when we add those two arguments to the ocr_predictor, it still tries to download files from https://storage.googleapis.com/tensorflow/keras-applications/resnet/resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5 and https://doctr-static.mindee.com/models?id=v0.4.1/vgg16_bn_r-c5836cea.zip&src=0    
Also when we pass in those arguments, the text extraction is no longer accurate and returns garbage values.

**Reply by felixdittrich92** [reply]
Could you share a code snippet how you load  the models and init the predictor please

**Reply by johnbuts** [reply]
Yeah sure: 
def extract_text_from_img(img):
    img_doc = DocumentFile.from_images(img.getvalue())
    model = ocr_predictor(pretrained=False, pretrained_backbone=False)
    result = model(img_doc)
    extracted_words = []
    for pages in result.pages:
        for blocks in pages.blocks:
            for lines in blocks.lines:
                for words in lines.words:
                    extracted_words.append(words.value)


And the output for an image: python DoctrTest.py
Fontconfig error: Cannot load default config file: No such file: (null)
°=Ï ÈZ °=ÏZ °=ÏZ °=ÏZ °ù=ÏZ °=ÏZ °=ÏZ °=ÏZ °=ÏZ °=ÏZ °=ÏZ °=ÏZ °=ÏZ °=ÏZ °=ÏZ °ÈÏZ °ÏZ °ÏZ

**Reply by felixdittrich92** [reply]
Hi @johnbuts :wave:,

Excuse the late response.

```
from doctr.models import ocr_predictor, db_resnet50, crnn_vgg16_bn

# Load custom detection model
det_model = db_resnet50(pretrained=False, pretrained_backbone=False)
det_model.load_weights("<path_to_checkpoint>/weights")

# Load custom recognition model
reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False)
reco_model.load_weights("<path_to_checkpoint>/weights")

# Init predictor
predictor = ocr_predictor(det_arch=det_model, reco_arch=reco_model, pretrained=False, pretrained_backbone=False)
```

In this case you avoid downloading the models

Best regards,
Felix

**Reply by johnbuts** [reply]
Hello @felixdittrich92 
No worries on the response time
So I believe there is some confusion on the load_weights lines of code. The zip file that we downloaded is attached below, and the checkpoints file is just a file with some text inside of it, and doesn't contain another directory named weights within it. We were wondering if you could maybe point us in the right direction to download the checkpoint/weights file. We also tried replacing the checkpoint/weights with the other weight files within the zip and it also led to no success. 

zip link:  http://doctr-static.mindee.com/models?id=v0.2.0/db_resnet50-adcafc63.zip&src=0

Thanks!
Johnbuts

**Reply by felixT2K** [reply]
Hey :)

1. download the zip file
2. extract the archive (you will get a folder which contains a checkpoint, a weights.data-xx and a weights.index file)
3. `path = <YOUR_UNZIPPED_FOLDER_PATH>/weights`

You are correct there is not exactly a weights folder but TF handles resolving the weights files internally

Example:

`path = "/home/felix/.cache/doctr/models/db_resnet50-adcafc63/weights"`

**Reply by johnbuts** [reply]
So this is the code we're running: 

    img_doc = DocumentFile.from_images(img.getvalue())
    det_model = db_resnet50(pretrained=False, pretrained_backbone=False)
    det_model.load_weights(r"C:\Users\Downloads\db_resnet50-adcafc63\weights")
    reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False)
    reco_model.load_weights(r"C:\Users\Downloads\vgg16_bn_r-c5836cea\weights")
    model = ocr_predictor(det_arch=det_model, reco_arch=reco_model, pretrained=False, pretrained_backbone=False)
    result = model(img_doc)

On the reco_model.load_weights line of code, we're getting the following error:
 In this package: site-packages\tensorflow\python\checkpoint\checkpoint.py    on line 886: 
``` 
Exception has occurred: AssertionError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
Nothing except the root object matched a checkpointed value. Typically this means that the checkpoint does not match the Python program. The following objects have no matching checkpointed value: [<tf.Variable 'batch_normalization_13/gamma:0' shape=(256,) dtype=float32, numpy=
array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1.], dtype=float32)>, <tf.Variable 'batch_normalization_15/moving_mean:0' shape=(512,) dtype=float32, numpy=
array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0.], dtype=float32)>, <tf.Variable 'conv2d_20/kernel:0' shape=(3, 3, 512, 512) dtype=float32, numpy=
array([[[[-0.02684   , -0.02769584,  0.0086169 , ..., -0.00184662,
           0.03285538, -0.01932777],
         [-0.02337521,  0.0042742 , -0.02859647, ..., -0.01599606,
          -0.03713632, -0.00856926],
         [-0.02956581,  0.00440462,  0.00320838, ...,  0.03110453,
           0.00462055, -0.01867752],
         ...,
         [-0.00415884, -0.04231287,  0.00941052, ...,  0.02585342,
           0.00413524, -0.03338748],
         [ 0.01639786,  0.02487399,  0.01622463, ...,  0.02431566,
          -0.01099479,  0.00278823],
         [ 0.0080782 , -0.04451105,  0.00239585, ..., -0.0179261 ,
           0.01031822,  0.00038793]],

        [[-0.02468134, -0.03938416,  0.02506799, ...,  0.02986104,
          -0.03002295,  0.02028053],
         [-0.01176625, -0.0163557 ,  0.01441632, ..., -0.03174131,
           0.01746994,  0.03471561],
         [ 0.03362954,  0.00497933, -0.02659766, ..., -0.0393213 ,
          -0.02077404,  0.01060349],
         ...,
         [ 0.04450123, -0.01821576,  0.03668294, ..., -0.00054244,
           0.02350906,  0.00291506],
         [ 0.02119054,  0.01217214,  0.01207616, ...,  0.01066085,
           0.02013449,  0.0398267 ],
         [ 0.0192862 , -0.00943688,  0.00533575, ..., -0.02814786,
          -0.03662217, -0.01360207]],

        [[-0.00281842,  0.00652939,  0.0248386 , ...,  0.00802593,
          -0.00650645,  0.0082506 ],
         [-0.01502103, -0.03927404,  0.02144871, ...,  0.01081521,
           0.02957435, -0.02565934],
         [ 0.01956783, -0.00277034, -0.02852971, ..., -0.01411765,
          -0.00587691, -0.01535598],
         ...,
         [-0.02568277, -0.02688205,  0.0198041 , ...,  0.02072545,
          -0.02094079,  0.00871276],
         [ 0.01636026, -0.0143536 ,  0.0197771 , ...,  0.02608879,
          -0.02881513,  0.00148486],
         [ 0.02059323, -0.03948075,  0.02483892, ...,  0.01686582,
          -0.00532641, -0.0004546 ]]],


       [[[-0.02577316,  0.04114323, -0.00720073, ..., -0.0111932 ,
           0.01232223,  0.04489302],
         [ 0.00957667, -0.02793128, -0.02636868, ..., -0.02805854,
           0.03250508,  0.01905689],
         [-0.00520651, -0.01243506,  0.00793857, ..., -0.00670214,
           0.00600238,  0.00075862],
         ...,
         [ 0.01246245,  0.04353509,  0.00585921, ..., -0.01175412,
          -0.00607807,  0.01715075],
         [-0.02806814, -0.0318185 ,  0.01340313, ..., -0.01920047,
          -0.04646638, -0.00293118],
         [-0.00373556, -0.03833161, -0.00488808, ..., -0.00045666,
          -0.02944783, -0.01329148]],

        [[ 0.00201686,  0.00503103,  0.02070234, ..., -0.00717057,
           0.02384776, -0.02505942],
         [ 0.01878778,  0.01742364,  0.02027432, ...,  0.02596615,
          -0.03585054,  0.00859969],
         [-0.02935585, -0.02231303,  0.00204076, ...,  0.01870481,
           0.01699754,  0.02045069],
         ...,
         [-0.01649569,  0.0153031 , -0.01008522, ...,  0.01582802,
           0.01234656, -0.01232189],
         [ 0.01045115,  0.01620776,  0.01599875, ...,  0.0009209 ,
          -0.00016659, -0.00765404],
         [ 0.0143618 ,  0.02854638, -0.01734245, ...,  0.00219031,
           0.01747672,  0.00967824]],

        [[-0.0186919 ,  0.01046404,  0.00042238, ..., -0.02398162,
           0.00431164, -0.00409181],
         [-0.01372487,  0.00914704, -0.00364027, ...,  0.00181408,
          -0.03106484,  0.00749677],
         [ 0.01622033,  0.01738609, -0.00011113, ..., -0.01239515,
           0.00184304, -0.02309938],
         ...,
         [-0.00465053, -0.00221076, -0.01308466, ...,  0.01130565,
           0.02548616, -0.02170796],
         [ 0.00236461,  0.00042414,  0.00763269, ..., -0.02713031,
          -0.00897902,  0.01966306],
         [ 0.01303527,  0.01966631, -0.02929495, ..., -0.01657318,
           0.01738826, -0.00803662]]],


       [[[ 0.0168044 ,  0.01072365, -0.00960741, ..., -0.03021759,
          -0.02501692, -0.01114801],
         [-0.00810971, -0.00689367,  0.02944706, ...,  0.00236321,
           0.02984924,  0.03055329],
         [-0.0048275 ,  0.004876  , -0.00231964, ...,  0.00747669,
           0.00222679,  0.01353622],
         ...,
         [-0.01274361, -0.01051659,  0.01292477, ...,  0.02519551,
           0.00049164,  0.00186732],
         [ 0.0209384 ,  0.01631491, -0.04191035, ..., -0.03069601,
          -0.00683762, -0.00958218],
         [-0.04108559, -0.02339591, -0.04010142, ..., -0.01184398,
          -0.00469683, -0.00521049]],

        [[-0.02783923, -0.00899031, -0.00798697, ..., -0.00959796,
           0.00512867,  0.01190163],
         [ 0.00932257,  0.00889392,  0.00126547, ..., -0.00100358,
          -0.02199283, -0.0219455 ],
         [ 0.0018891 , -0.00896772, -0.01676869, ..., -0.03238862,
          -0.01819157,  0.04231785],
         ...,
         [ 0.00896939, -0.00860165,  0.0203834 , ...,  0.00444824,
           0.01037745, -0.01557526],
         [-0.02576968, -0.00050193, -0.00725488, ..., -0.00664695,
          -0.00367506,  0.02941673],
         [ 0.03750513, -0.00540059, -0.01496672, ...,  0.01038816,
          -0.00354234,  0.03654788]],

        [[ 0.02579843,  0.03048592,  0.02272199, ...,  0.02199354,
          -0.02505499, -0.00035134],
         [ 0.00673714, -0.00130881, -0.00629322, ...,  0.00332529,
          -0.01122589,  0.0115717 ],
         [-0.00045848, -0.00701221,  0.0177513 , ...,  0.00855703,
          -0.02971721, -0.03124617],
         ...,
         [ 0.00119128, -0.00053088,  0.00079367, ...,  0.02103709,
          -0.00044379, -0.01153346],
         [ 0.00700835, -0.01277481, -0.04546491, ..., -0.00843005,
          -0.01385275, -0.01205609],
         [ 0.02818107,  0.00341262,  0.02746619, ...,  0.02525842,
           0.01176539,  0.03620316]]]], dtype=float32)>, <tf.Variable 'batch_normalization_8/gamma:0' shape=(64,) dtype=float32, numpy=
array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.], dtype=float32)>, <tf.Variable 'batch_normalization_15/moving_variance:0' shape=(512,) dtype=float32, numpy=
array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1.], dtype=float32)>, <tf.Variable 'batch_normalization_10/gamma:0' shape=(128,) dtype=float32, numpy=
array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1.], dtype=float32)>, <tf.Variable 'bidirectional/backward_lstm/lstm_cell/kernel:0' shape=(512, 512) dtype=float32, numpy=
array([[ 0.05115586, -0.0215711 , -0.05692567, ..., -0.01208837,
         0.04989385, -0.01082986],
       [ 0.00759807,  0.06224047, -0.04096587, ..., -0.00209692,
        -0.03073034, -0.0033906 ],
       [-0.00202228,  0.0176376 , -0.05057842, ..., -0.06828018,
         0.02579145,  0.04850743],
       ...,
       [-0.01552839, -0.0389268 ,  0.02952904, ...,  0.03301635,
        -0.00820381, -0.04494096],
       [-0.02572073,  0.02546196,  0.05069869, ..., -0.01050298,
         0.0233073 ,  0.07380352],
       [-0.03133208,  0.02393896,  0.06595797, ..., -0.00651816,
        -0.02473846, -0.02213143]], dtype=float32)>, <tf.Variable 'batch_normalization_16/beta:0' shape=(512,) dtype=float32, numpy=
```

the error continues to print out a bunch of more weight values, but you get the idea. I suspect we have the wrong crnn_vgg16_bn package, we also tried using the db_resnet50-adcafc63\weights for the crnn_vgg16_bn load weights but was left with a similar error

Thanks,
Johnbuts

**Reply by felixdittrich92** [reply]
You have Download the vgg backbone classification model 
The correct URL is https://doctr-static.mindee.com/models?id=v0.3.0/crnn_vgg16_bn-76b7f2c6.zip&src=0

Code looks good now only download the correct one and change the path :)

**Reply by johnbuts** [reply]
Sorry for the late reply
The new url you gave us + the code snippets god rid of the http error (WOOHOO!!! THANKS), but now we're getting this error below, we were wondering what we could do differently. We thought this was just a warning before, but the code stops executing.

2024-01-26 20:35:21.356175: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-01-26 20:35:21.359430: I external/local_tsl/tsl/cuda/cudart_stub.cc:31] Could not find cuda drivers on your machine, GPU will not be used.
2024-01-26 20:35:21.398792: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-01-26 20:35:21.398849: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-01-26 20:35:21.400078: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-01-26 20:35:21.406789: I external/local_tsl/tsl/cuda/cudart_stub.cc:31] Could not find cuda drivers on your machine, GPU will not be used.
2024-01-26 20:35:21.407416: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-01-26 20:35:22.408361: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT

**Reply by felixdittrich92** [reply]
That's warnings from TensorFlow (comes not from doctr) if you run on CPU (i have the same on my machine ^^)
You can ignore it or disable the warnings

**Reply by felixdittrich92** [reply]
oh sorry I missed it, it stops ? without any further error traceback ? 🤔

**Reply by johnbuts** [reply]
Turns out we weren't allocating enough memory to our container... But now it works. Thanks so much for the help, you're a stellar developer!!!

**Reply by felixdittrich92** [reply]
Haha nice that it works now :hugs: Thanks :sweat_smile:
